File: Leetcode_Python/simplify-path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        
        # "." -> current dir
        #  ".." -> prev dir
        #  "//" or. "///" -> "/"
        # "..." -> dir

        # MUST START WITH "/" 
        # MUST BE SEPARATED BY "/"
        # CANNOT END WITH "/" except for root
        # CANNOT HAVE "." or ".."

        stack = list()
        dir = path.split("/")

        for el in dir:
            if el == "." or not el:
                continue
            elif el == "..":
                if stack:
                    stack.pop()
            else:
                stack.append(el)
        return "/" + "/".join(stack)

File: Leetcode_TS/simplify-path.ts
function simplifyPath(path: string): string {
    let stack: string[] = []
    let splitPath = path.split("/")

    for (let i = 0; i < splitPath.length; i++) {
        if (!splitPath[i] || splitPath[i] == ".") continue

        else if (splitPath[i] == "..") stack.pop()

        else {
            stack.push(splitPath[i])
        }
    }

    return "/" + stack.join("/")

    
};